cfgov/v1/models/filterable_page.py

```diff
@@ -13,6 +13,19 @@
 from v1.util.ref import get_category_children
 
 
+class SearchResultsPaginator(Paginator):
```

Comment:
This logic lets us move the pagination upstream into the Elasticsearch query instead of pulling back the full list of page IDs from ES and paginating that. I dug into the django-opensearch-dsl code and there doesn't seem to be anything built-in to support this.

Adding `prefetch_related` here also lets us prefetch the page related objects to avoid the [N+1 queries](https://scoutapm.com/blog/django-and-the-n1-queries-problem) problem.